backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -680,6 +690,13 @@ export class SegmentService {
     const collectErrors = async (segment: SegmentInputValidator, contexts: string[], isList: boolean) => {
       let errorMessage = '';
       let compatibilityType = IMPORT_COMPATIBILITY_TYPE.COMPATIBLE;
+
+      const contextValidationError = this.validateSegmentContext(segment);
+      if (contextValidationError) {
+        errorMessage += contextValidationError;
+        compatibilityType = IMPORT_COMPATIBILITY_TYPE.INCOMPATIBLE;
+      }
+
```

Comment:
If the context doesn't exist, we shouldn't look for group types on line 747 below. Maybe combine the `if`s from lines 695 and 742 and add the error message there.

Reply:
Right, I think `if (!contexts.includes(segment.context)) {` is sufficient and I didn't need to add the additional validation there. This file probably could have remained unchanged. I will verify this and make a commit.

Reply:
I verified that the context validation works fine even without any changes. I've reverted the changes I made to the `SegmentService.ts`.

Reply:
This still throws a 500 when the context doesn't exist, because if looks for group types in the non-existent context. I'd change line 725 from `     if (!contexts.includes(segment.context)) {` to `       if (!contexts.includes(segment.context) || !contextMetaDataOptions.includes(segment.context)) {
`

Reply:
I applied the changes as we discussed.